frontend/src/components/message/useMessage.ts

```diff
@@ -0,0 +1,67 @@
+import { KeyboardEvent, useRef, useState } from "react";
+
+const useMessage = () => {
+    const [selected, setSelected] = useState("");
+    const [isEditing, setEditing] = useState(false);
+    const [currentMessage, setCurrentMessage] = useState("");
+    const messageInput = useRef<HTMLInputElement>(null!);
+
+    const input = messageInput.current;
+
+    const sendRequest = () => {
+        //TODO: implement
+    };
+
+    const messageHasChanged = () => input.textContent !== currentMessage;
+
+    const handleAcceptMessage = (event: KeyboardEvent<HTMLInputElement>) => {
+        if (!(event.key === "Enter" || event.key === "Escape")) {
+            return;
+        }
+
+        input.contentEditable = "false";
+
+        if (messageHasChanged()) {
+            sendRequest();
+        }
+
+        setEditing(false);
```

Comment:
I think this state is unnecessary, because u can use a value of contentEditable from ref